### ISSUE TITLE
any documents about the output module of Polarizability

### ISSUE THREAD

**Issue by lixinyuu** [atomistic-machine-learning/schnetpack#178]
Hi,

The output module of Polarizability looks very interesting. However, I cannot get this run with the following code.
```
import os
import logging
import torch
from torch.optim import Adam
import schnetpack as spk
import schnetpack.representation as rep
from schnetpack.train import Trainer, CSVHook, ReduceLROnPlateauHook
from schnetpack.train.metrics import MeanAbsoluteError
from schnetpack.train import build_mse_loss
import torch
import torch.nn as nn
import torch.nn.functional as F
from schnetpack.nn.base import Dense
from schnetpack.nn.cfconv import CFConv
from schnetpack.nn.cutoff import HardCutoff
from schnetpack.nn.neighbors import AtomDistances
from schnetpack.nn.activations import shifted_softplus
from schnetpack.representation.schnet import SchNetInteraction
import shutil
from schnetpack.nn.base import Aggregate
model_dir = "schnet_polar"
if torch.cuda.is_available():
    print("running on GPU")
    
os.makedirs(model_dir)
properties = ["isotropic_polarizability"]
logging.info("the available properties are {}".format(properties[0]))
logging.info("get dataset")

dataset = spk.AtomsData("/tmp/qm9-10-6.db", load_only=properties)
train, val, test = spk.train_test_split(
    dataset, 100000, 10000, os.path.join(model_dir, "split.npz")
)
train_loader = spk.AtomsLoader(train, batch_size=64, shuffle=True)
val_loader = spk.AtomsLoader(val, batch_size=64)
means, stddevs = train_loader.get_statistics(properties)
print(means, stddevs)

# model build
logging.info("build model")
representation = rep.SchNet(n_interactions=6, cutoff = 10)
output_modules = [spk.atomistic.Polarizability(n_in = 128, isotropic=True, property=n) for n in properties ]
model = spk.atomistic.AtomisticModel(representation, output_modules)

# build optimizer
optimizer = Adam(model.parameters(), lr=1e-4)

# hooks
logging.info("build trainer")
metrics = [MeanAbsoluteError(p, p) for p in properties]
hooks = [CSVHook(log_path=model_dir, metrics=metrics), ReduceLROnPlateauHook(optimizer)]

# trainer
loss = build_mse_loss(properties)
trainer = Trainer(
    model_dir,
    model=model,
    hooks=hooks,
    loss_fn=loss,
    optimizer=optimizer,
    train_loader=train_loader,
    validation_loader=val_loader,
)

# run training
logging.info("training")
trainer.train(device='cuda')
```
The error message as follows
```
Traceback (most recent call last):
  File "sch-polar.py", line 79, in <module>
    trainer.train(device='cpu')
  File "/users/pythonenv/pytorch/lib/python3.6/site-packages/schnetpack-0.3-py3.6.egg/schnetpack/train/trainer.py", line 263, in train
  File "/users/pythonenv/pytorch/lib/python3.6/site-packages/schnetpack-0.3-py3.6.egg/schnetpack/train/trainer.py", line 190, in train
  File "/users/pythonenv/pytorch/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/users/pythonenv/pytorch/lib/python3.6/site-packages/schnetpack-0.3-py3.6.egg/schnetpack/atomistic/model.py", line 44, in forward
  File "/users/pythonenv/pytorch/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/users/pythonenv/pytorch/lib/python3.6/site-packages/schnetpack-0.3-py3.6.egg/schnetpack/atomistic/output_modules.py", line 467, in forward
IndexError: index 1 is out of bounds for dimension 3 with size 1
```
Could you please give me some suggestions about this? By the way, is there any reference about this Polarizability? Sorry I can not understand what this is trying to do, thus have no idea how to debug this.
Thanks.

**Comment by XinyuuLi** [comment]
More information about this issue. I think the error come from the following line

https://github.com/atomistic-machine-learning/schnetpack/blob/ae0c781dcbb4048d40d7b9fdc68a67bbb13efe00/src/schnetpack/atomistic/output_modules.py#L461
neighbor_contributions is a B*A*N*1 matrix instead of B*A*N*2. This line throw error message as 
"neighbor_contributions2 = neighbor_contributions[:, :, :, 1]
IndexError: index 1 is out of bounds for dimension 3 with size 1"

Thanks.

**Comment by Stefaanhess** [comment]
Hi @XinyuuLi ,
You are right, there seems to be an issue with the dimensions.
Maybe you can use the spk.atomistic.Atomwise output module, which has also been used for the benchmark results, while we are fixing the issue.

**Comment by XinyuuLi** [comment]
Thanks @Stefaanhess , 
I am quite interested in this output module. It looks like you are going to calculate the  deriative of dipole on the field direction. Have you done any test with this output module? BTW, it is appreciated if you can add the math formulas here. Thanks again.

**Comment by Stefaanhess** [comment]
I will link @mgastegger here, who is unfortunately on holliday until next week. He should be able to tell you more about the output module.

**Comment by Stefaanhess** [comment]
Hey @XinyuuLi ,

the polarizability-output shuold be fixed with #179 . I will test it on QM9 and add it to the benchmarks.
